### PR TITLE
fix(db): self-heal connection pool and fix native adapter handle leak

### LINKED ISSUE #25: Connection pool never evicts dead connections; one break is permanent until process restart
### Summary

`ConnectionPool` in `src/db/connection.ts` reuses cached connections blindly. Once any pooled connection breaks, it stays in rotation forever and every subsequent query that lands on it fails. The server "loses" the database and stays broken until the process is restarted, even though a fresh `isql` connection works fine.

### Environment

- Firebird 3.0 server (no server-side idle timeout, no statement timeout; those are 4.0+, so the server never auto-cleans stale/leaked connections. The client must self-heal).
- Native driver path (`--use-native-driver` / `node-firebird-driver-native`), required for WireCrypt.
- Read-only DB user.

### Root cause

- `acquire()` does `this.pool.pop()` with no liveness check and hands back whatever is cached.
- `detach` is monkey-patched to `release()`, which pushes the connection back into the pool even after a failed query.
- `executeQuery` (`src/db/queries.ts`) calls `db.detach()` in its `finally`, so a connection that just failed is returned to the pool and handed to the next caller.
- Pool size is hardcoded (`new ConnectionPool(config, 5)`), and `release()` only truly disconnects when the pool is full, so connections effectively never close.

Net effect: a connection only has to break once (idle socket drop from firewall/NAT/conntrack with no keepalive, a one-off server restart / nightly gbak / admin kill, or a poisoned attachment) and it is stuck in the pool permanently.

### Steps to reproduce

1. Start the server against a Firebird 3.0 DB using the native driver.
2. Run a query to warm the pool.
3. Force a mid-life drop on the pooled connection: `DELETE FROM MON$ATTACHMENTS` for that session, restart Firebird, or drop the TCP path with a firewall rule / tcpkill.
4. Run another query.

Expected: the server transparently reconnects.
Actual: queries error and keep erroring until the process is restarted.

### Proposed fix

1. Validate-on-acquire: before returning a pooled connection, run a cheap read-only probe (`SELECT 1 FROM RDB$DATABASE`) through the same adapter. On failure, really disconnect, decrement the active count, and try the next/new connection (handle several stale entries in a row).
2. Evict-on-error, do not recycle: give the pool explicit `release(db)` (healthy, back to pool) and `destroy(db)` (dead, real disconnect + active-count decrement + serve any waiter with a fresh connection). `executeQuery` calls `destroy` on a caught error, `release` on success.
3. Idle reaper / max-age: stamp each pooled connection with `lastUsed` and evict pooled connections older than an idle TTL, reopening fresh. This is the durable protection on the native path where `SO_KEEPALIVE` cannot be set from JS.
4. Make it configurable: `FIREBIRD_POOL_MAX` (default 5) and `FIREBIRD_POOL_IDLE_MS` (default 60000, `0` disables the reaper), so a deployment running many server processes (one per database) can bound total idle connections by setting the max to 1 or 2.

### Verification

- Repro above transparently reconnects after the fix.
- With the reaper: idle past the TTL, then query. The old connection is reaped, a fresh one is used, and the `MON$ATTACHMENTS` count stays low.

I have a fix implemented and tested for this and am happy to open a PR.

### LINKED ISSUE #26: Native driver adapter leaks statement/result-set handles on the error path
### Summary

In `NativeDriver.createAdapter().query` (`src/db/driver-factory.ts`), `resultSet.close()` and `statement.dispose()` run only on the success path. On any error after `prepare`, the statement (and the result set, if `fetchAsObject` threw) is never disposed. Those handles leak on the attachment, which is reused across every query, so repeated query errors progressively degrade the shared attachment until every operation on it fails. No network event is required to trigger this.

### Root cause

```ts
const statement = await attachment.prepare(transaction, sql);
let rows = [];
if (statement.hasResultSet) {
    const resultSet = await statement.executeQuery(transaction, params);
    rows = await resultSet.fetchAsObject();
    await resultSet.close();   // success path only
} else {
    await statement.execute(transaction, params);
}
await statement.dispose();     // success path only
await transaction.commit();
```

The `catch` block only rolls back the transaction (and even that is merely logged if `rollback()` itself throws). `statement` and `resultSet` are declared inside the `try`, so the error path cannot free them.

### Steps to reproduce

1. Run a batch of failing queries against a native-driver connection (for example, a series of syntax errors on the same pooled connection).
2. Run a valid query on that same connection.

Expected: the valid query succeeds; `MON$ATTACHMENTS` shows no accumulating statements/transactions for that session.
Actual: handles accumulate on the reused attachment and operations begin to fail.

### Proposed fix

Move `resultSet.close()` and `statement.dispose()` into a `finally` so handles are released on both the success and error paths (guarding each release in its own try/catch and logging failures). On the success path, free them in order and clear the references so `finally` does not attempt a double free (closing a result set after `commit` would otherwise throw and spam the logs every query).

### Verification

- Run many failing queries (syntax errors), then a valid one: the valid query still works.
- `MON$ATTACHMENTS` shows no accumulating statements/transactions after the barrage of errors.

I have a fix implemented and tested for this and am happy to open a PR. It pairs with the connection-pool eviction issue, but is independent: this leak degrades a connection with no network event, while the pool bug makes any single break permanent.

### PR DESCRIPTION
**Pull request by tomaskir**
## Summary

Fixes two independent bugs that make the server intermittently "lose" the database and stay broken until the process is restarted, observed on Firebird 3.0 with the native driver (WireCrypt).

Fixes #25
Fixes #26

## Connection pool never evicts dead connections (#25)

`ConnectionPool` in `src/db/connection.ts` reused cached connections blindly, so once any pooled connection broke it stayed in rotation forever. Firebird 3.0 has no server-side idle or statement timeout, so nothing cleans up a stale connection; the client has to self-heal.

- Validate-on-acquire: probe pooled connections with a read-only `SELECT 1 FROM RDB$DATABASE` through the same adapter before handing them back; on failure, really disconnect and reopen, handling several stale entries in a row.
- Evict-on-error: replaced the "release everything" detach hook with explicit `release(db)` (healthy, back to pool) and `destroy(db)` (dead, real disconnect + active-count decrement + serve any waiter with a fresh connection). `executeQuery` now destroys on error and releases on success, instead of returning a just-failed connection to the pool.
- Idle reaper: stamp `lastUsed` and reap pooled connections older than an idle TTL, then reopen fresh. This is the durable protection on the native path, where `SO_KEEPALIVE` cannot be set from JS.
- Configurable: `FIREBIRD_POOL_MAX` (default 5) and `FIREBIRD_POOL_IDLE_MS` (default 60000, `0` disables the reaper), so a deployment running many server processes (one per database) can bound total idle connections.

## Native adapter leaks statement/result-set handles on error (#26)

In `NativeDriver.createAdapter().query` (`src/db/driver-factory.ts`), `resultSet.close()` and `statement.dispose()` ran only on the success path. On any error after `prepare`, those handles leaked on the reused attachment until every operation on it failed, with no network event required.

- Moved `resultSet.close()` / `statement.dispose()` into a `finally` so handles are freed on the error path too. On success they are freed in order and the references cleared, so `finally` does not double-free (closing a result set after `commit` would otherwise throw on every query).

## Tests

- Added `src/__tests__/db/connection-pool.test.ts` covering healthy reuse, stale eviction (single and several in a row), the idle reaper (and its disable via `idleMs = 0`), `destroy` freeing a slot, and a waiter being served a fresh connection after `destroy`.
- `jest.config.js`: run tests as CommonJS with a `.js`-extension `moduleNameMapper` so the ESM (`NodeNext`) source can be imported under ts-jest, and match only `.ts` test files (skips stale compiled `.js` artifacts).

`npm run build` is clean and the full jest suite passes (18/18).

## Notes / scope

- The pure-JS `node-firebird` path is preserved; probe, release, and destroy all go through the shared adapter interface.
- `resources/events.ts` attaches directly via the driver (bypassing the pool), so proactive events are unaffected.
- These two fixes are independent and could be reviewed or merged separately, but they share the same symptom and the same files, so they are bundled here.

### CODE REVIEW

**Comment by PuroDelphi** [comment]
Thank you so much for this amazing contribution! Your root-cause analysis is spot on. We have manually merged your PR into our alpha branch (v2.9.0-alpha.5) where it successfully passed all our tests and will be part of the upcoming release. Closing this PR as the code has already been integrated. Great work!
